static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -160,33 +134,31 @@ function useReplayData({replaySlug, orgSlug}: Options): Result {
     return response.data;
   }, [api, orgSlug, projectSlug, replayId]);
 
-  const fetchSegmentList = useCallback(async () => {
-    const response = await api.requestPromise(
-      `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/recording-segments/`
-    );
-    return response.data as ReplaySegment[];
-  }, [api, orgSlug, projectSlug, replayId]);
+  const fetchAllRRwebEvents = useCallback(async () => {
+    let next: ParsedHeader = {
+      href: `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/recording-segments/?download&per_page=5`,
+      results: true,
+      cursor: '',
+    };
 
-  const fetchRRWebEvents = useCallback(
-    async (segmentIds: number[]) => {
-      const attachments = await Promise.all(
-        segmentIds.map(async segmentId => {
-          const response = await api.requestPromise(
-            `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/recording-segments/${segmentId}/?download`,
-            {
-              includeAllArgs: true,
-            }
-          );
-
-          return decompressSegmentData(...response);
-        })
-      );
-
-      // ReplayAttachment[] => ReplayAttachment (merge each key of ReplayAttachment)
-      return flattenListOfObjects(attachments);
-    },
-    [api, replayId, orgSlug, projectSlug]
-  );
+    const segmentRanges: any = [];
+    while (next.results) {
```

Comment:
i think this is fine for now, but we can probably do it in parallel by first determining the number of segments then splitting them up. or if we simply returned the first set then did the rest in the background while the initial page loaded

Reply:
yeah, it could be smarter like that. We will have to test how things work when data streams in, but it should be limited to changes in this file and in our ReplayReader class

Can be queued up when we're looking at network perf/optimizations even more.

Reply:
added.